src/main/java/org/fenixedu/academic/domain/Person.java

```diff
@@ -219,6 +219,11 @@ public void setGivenNames(String newGivenNames) {
         profile.changeName(newGivenNames, profile.getFamilyNames(), profile.getDisplayName());
     }
 
+    public void setNickname(String newNickname) {
```

Comment:
This changes the displayName of the UserProfile. Shouldn't the setter have a similar name ?